modules/GenericDataParser/src/server/generic_data_parser.R

```diff
@@ -706,27 +706,58 @@ validateCalculatedResults <- function(calculatedResults, dryRun, curveNames, tes
+        if(!"type" %in% colnames(projectsDT)) {
+          projectsDT[ , type := NA]
+        }
+
+        # Get the project type of the experiment project if required by configurations
+        projectType <- NA
+        if(configList$server.project.type.required) {
+          projectType <- projectsDT[code == projectCode]$type
+          if(is.na(projectType)) {
+            stopUser(paste0("Project '",projectCode,"' does not have a type. Please contact your system administrator."))
+          }
+        }
+
+        # Here we are annotating each of the projects with a boolean as to whether it can be used in the experiment if its used by a lot     
         projectsDT[ , projectAllowedForExperiment := {
-          # Check if the project is unrestricted or if it equals the projectCode of the experiment
-          !isRestricted | code == projectCode
+          # projectType and projectCode are referring to the experiment project
+          # code and isRestricted are referring to the batch project
+          if(is.na(projectType)) {
+            # If the project type is na then this is not a system with project type
+            # so we just check if the project is restricted and if the lot project is the same as the experiment project
+            !isRestricted | code == projectCode
+          } else {
+            # If project type is not na then this is a system with project type
+            switch(projectType,
+                    "RESTRICTED" =  !isRestricted | code == projectCode,
+                    "HYPER_RESTRICTED" = code == projectCode,
+                    "UNRESTRICTED" = TRUE,
+                    "GLOBAL" = ifelse(identical(type, "HYPER_RESTRICTED"), FALSE, TRUE),
+                    stopUser(paste0("Project '",projectCode,"' has an invalid type. Please contact your system administrator."))
+            )
+          }
+
         }, by = code]
 
         # Merge batch projects to project restrition information
         batchProjectWithRestrictionInfo <- as.data.table(merge(batchProjects, projectsDT, by.x="Project.Code", by.y="code"))
-        
+
         # Create a data table with one row per batch with a boolean column as to whether it can be used in the experiment
         # if any of the projects the batch belongs to is allowed to be loaded to this experiment, then the batch can be loaded to the experiment
-        canUseBatchDT <- batchProjectWithRestrictionInfo[ , any(projectAllowedForExperiment), by = Requested.Name]
+        canUseBatchDT <- batchProjectWithRestrictionInfo[ , any(projectAllowedForExperiment), by = c("Requested.Name", "type")]
         setnames(canUseBatchDT, "V1", "batchCanBeLoadedToExperimentProject")
 
         # Get a list of batches which are restricted and cannot be used in this experiment's project
-        rCompounds <- canUseBatchDT[batchCanBeLoadedToExperimentProject == FALSE]$Requested.Name
+        rCompounds <- canUseBatchDT[batchCanBeLoadedToExperimentProject == FALSE]
```

Comment:
Needed to keep the entire data table around so I could use it later on to produce the right messages per project type.